### PR TITLE
Refactor file attachment repositories

### PR DESCRIPTION
**Pull request by AndriiHorishniiMOC**
**Jira issue:**
https://glia.atlassian.net/browse/MOB-3791

**What was solved?**
Remove `SecureFileAttachmentRepository` and some SC use cases that duplicated the logic.
Refactor `FileAttachmentRepository` and fileupload use cases. `FileAttachmentRepository` is used for both SC and Live Engagement.

**Release notes:**

 - [ ] Feature
 - [ ] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [ ] Is the feature sufficiently tested? All tests fixed? Necessary unit, acceptance, snapshots added? Check that at least new public classes & methods are covered with unit tests
 - [ ] Did you add logging beneficial for troubleshooting of customer issues?
 - [ ] **Did you add new logging?** We would like the logging between platforms to be similar. Refer to [**Logging from Android SDKs** → **Things to consider for newly added logs**](https://glia.atlassian.net/wiki/spaces/ENG/pages/3568861468/Logging+from+Android+SDKs#Things-to-consider-for-newly-added-logs) in Confluence for more information.

**Screenshots:**

### CODE REVIEW

File: widgetssdk/src/test/java/com/glia/widgets/core/fileupload/FileAttachmentRepositoryTest.kt [hunk 1]
```diff
@@ -324,87 +324,87 @@ class LocalAttachmentRepositoryTest {
...
+//    }
+
+//    @Test
+//    fun `addObserver calls update once when same observer added twice`() {
+//        clearMocks(observer1)
+//        subjectUnderTest.clearObservers()
+//
+//        subjectUnderTest.addObserver(observer1)
+//        subjectUnderTest.addObserver(observer1)
+//
```
Comment (andrews-moc) [inline widgetssdk/src/test/java/com/glia/widgets/core/fileupload/FileAttachmentRepositoryTest.kt:348]: I see several disabled tests. Is it a leftover?

**Review by andrews-moc — APPROVED** [review]
Looks good to me.

File: widgetssdk/src/test/java/com/glia/widgets/core/fileupload/FileAttachmentRepositoryTest.kt (hunk 1 above)
Comment (AndriiHorishniiMOC) [reply widgetssdk/src/test/java/com/glia/widgets/core/fileupload/FileAttachmentRepositoryTest.kt:348]: Oh, yes, we don't need those tests.

**Review by gugalo — APPROVED** [review]

**Comment by AndriiHorishniiMOC** [comment]
!squash

**Pull request merged**